### ISSUE TITLE
'match' func make clickhouse down

### ISSUE THREAD

**Issue by SuperEdison** [ClickHouse/ClickHouse#53910]
> You have to provide the following information whenever possible.

**Describe what's wrong**

#52403 
my clickhouse's version is 23.3.2.37 but i saw the latest 23.3version, this bug is still happen, and not fixed
> A clear and concise description of what works not as it is supposed to.

> A link to reproducer in [https://fiddle.clickhouse.com/](https://fiddle.clickhouse.com/).

**Does it reproduce on recent release?**

[The list of releases](https://github.com/ClickHouse/ClickHouse/blob/master/utils/list-versions/version_date.tsv)

**Enable crash reporting**

> If possible, change "enabled" to true in "send_crash_reports" section in `config.xml`:

```
<send_crash_reports>
        <!-- Changing <enabled> to true allows sending crash reports to -->
        <!-- the ClickHouse core developers team via Sentry https://sentry.io -->
        <enabled>false</enabled>
```

**How to reproduce**

* Which ClickHouse server version to use
* Which interface to use, if matters
* Non-default settings, if any
* `CREATE TABLE` statements for all tables involved
* Sample data for all these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary
* Queries to run that lead to unexpected result

**Expected behavior**

> A clear and concise description of what you expected to happen.

**Error message and/or stacktrace**

> If applicable, add screenshots to help explain your problem.

**Additional context**

> Add any other context about the problem here.

**Comment by SuperEdison** [comment]
23.3.11lts still is error, the clickhouse is  literally down

**Comment by alexey-milovidov** [comment]
Use the latest version.
Install it as follows:

```
curl https://clickhouse.com/ | sh
```

**Comment by SuperEdison** [comment]
> Use the latest version. Install it as follows:
> 
> ```
> curl https://clickhouse.com/ | sh
> ```

can u fix at the version of 23.3?
pls

**Comment by alexey-milovidov** [comment]
@SuperEdison, you can fix the version 23.3 by installing the version 23.7 on top of it.
